clients/cobol-lsp-vscode-extension/src/task/CustomBuildTaskTerminal.ts

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) 2022 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Broadcom, Inc. - initial API and implementation
+ */
+
+import { IProfileLoaded } from "@zowe/imperative";
+import { ZoweExplorerApi } from "@zowe/zowe-explorer-api";
+import * as fs from "fs";
+import * as path from "path";
+import * as vscode from "vscode";
+import { SettingsService } from "../services/Settings";
+import { COMPILE_JCL_HEADER, COMPILE_JCL_MIDDLE, COMPILE_JCL_TAIL } from "./script/compile";
+
+export class CustomBuildTaskTerminal implements vscode.Pseudoterminal {
+    private ERROR_REGEX = "^\\s+([0-9]*)\\s+IGY[A-Z]{2}\\d{4}-([A-Z])\\s+(.*)$";
+    private writeEmitter = new vscode.EventEmitter<string>();
+    onDidWrite: vscode.Event<string> = this.writeEmitter.event;
+    private closeEmitter = new vscode.EventEmitter<number>();
+    onDidClose?: vscode.Event<number> = this.closeEmitter.event;
+
+    constructor(private documentText: vscode.TextDocument, private definition: vscode.TaskDefinition) {
+    }
+
+    public open(_initialDimensions: vscode.TerminalDimensions | undefined): void {
+        this.doCompile();
+    }
+
+    public close(): void {
+    }
+
+    private async doCompile(): Promise<void> {
+        const zoweExplorerApi: ZoweExplorerApi.IApiRegisterClient = vscode.extensions.getExtension("Zowe.vscode-extension-for-zowe")?.exports;
+        let loadedProfile: IProfileLoaded;
+        if (!zoweExplorerApi) {
+            return Promise.reject("Please install zowe Explorer");
+        }
+        if (!(this.definition.steplib && this.definition.jobCard)) {
```

Comment:
We should make steplib optional (empty array or not defined at all). In such a case the linklist library concatenation is searched.

Reply:
Updated.